internal/rollup/terser-wrapped.js

```diff
@@ -74,51 +108,92 @@ function runterser(inputFile, outputFile, sourceMapFile) {
     },
     'mangle': !debug,
   };
+  return terserConfig;
+}
 
-  let config = configFile;
-  if (!config) {
-    config = tmp.fileSync({keep: false, postfix: '.json'}).name;
+/**
+ *
+ * @param {{inputFile: string, outputFile: string, sourceMapFile: string}[]} files
+ * @param {boolean} debug
+ */
+async function runTerserSerial(files, debug) {
+  const Terser = require(TERSER_MODULE_LOCATION);
+  for (const f of files) {
+    // run one set of terser at a time
+    await runTerser(f.inputFile, f.outputFile, f.sourceMapFile, Terser, debug);
   }
+}
 
-  fs.writeFileSync(config, JSON.stringify(terserConfig));
+/**
+ *
+ * @param {{inputFile: string, outputFile: string, sourceMapFile: string}[]} files
+ * @param {boolean} debug
+ */
+async function runTerserParallel(files, debug) {
+  // TODO: do some concurrency limiting here
```

Comment:
do we have to finish the TODO before we merge/release this?

Reply:
We should.  context switching will destroy performance gains.
`os.cpus().length - 1` is typically a good cap.

Reply:
I think a worker pool would also make sense here.  There's no need to tear down and bring up new threads and re-initialize terser for each file.